// vant/lib/steps/index.js
'use strict';

exports.__esModule = true;

var _utils = require('../utils');

exports.default = (0, _utils.create)({
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('div', { staticClass: "van-steps", class: "van-steps--" + _vm.direction }, [_vm.title || _vm.description ? _c('div', { staticClass: "van-steps__status" }, [_vm.icon || _vm.$slots.icon ? _c('div', { staticClass: "van-steps__icon" }, [_vm._t("icon", [_c('icon', { class: _vm.iconClass, attrs: { "name": _vm.icon } })])], 2) : _vm._e(), _c('div', { staticClass: "van-steps__message" }, [_c('div', { staticClass: "van-steps__title", domProps: { "textContent": _vm._s(_vm.title) } }), _c('div', { staticClass: "van-steps__desc", domProps: { "textContent": _vm._s(_vm.description) } })]), _vm._t("message-extra")], 2) : _vm._e(), _c('div', { staticClass: "van-steps__items", class: { 'van-steps__items--alone': !_vm.title && !_vm.description } }, [_vm._t("default")], 2)]);
  },

  name: 'van-steps',

  props: {
    icon: String,
    title: String,
    active: Number,
    iconClass: String,
    description: String,
    direction: {
      type: String,
      default: 'horizontal'
    },
    activeColor: {
      type: String,
      default: '#06bf04'
    }
  },

  data: function data() {
    return {
      steps: []
    };
  }
});
