// vant/lib/contact-card/index.js
'use strict';

exports.__esModule = true;

var _utils = require('../utils');

exports.default = (0, _utils.create)({
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('div', { staticClass: "van-contact-card", class: ["van-contact-card--" + _vm.type, { 'van-contact-card--uneditable': !_vm.editable }], on: { "click": _vm.onClick } }, [_c('div', { staticClass: "van-contact-card__content" }, [_vm.type === 'add' ? [_c('icon', { staticClass: "van-contact-card__icon", attrs: { "name": "add2" } }), _c('div', { staticClass: "van-contact-card__text" }, [_vm._v(_vm._s(_vm.addText || _vm.$t('addText')))])] : _vm.type === 'edit' ? [_c('icon', { staticClass: "van-contact-card__icon", attrs: { "name": "contact" } }), _c('div', { staticClass: "van-contact-card__text" }, [_c('div', [_vm._v(_vm._s(_vm.$t('name')) + "：" + _vm._s(_vm.name))]), _c('div', [_vm._v(_vm._s(_vm.$t('tel')) + "：" + _vm._s(_vm.tel))])])] : _vm._e()], 2), _vm.editable ? _c('icon', { staticClass: "van-contact-card__arrow", attrs: { "name": "arrow" } }) : _vm._e()], 1);
  },

  name: 'van-contact-card',

  props: {
    tel: String,
    name: String,
    addText: String,
    editable: {
      type: Boolean,
      default: true
    },
    type: {
      type: String,
      default: 'add'
    }
  },

  methods: {
    onClick: function onClick(event) {
      if (this.editable) {
        this.$emit('click', event);
      }
    }
  }
});
