// vant/lib/mixins/popup/index.js
'use strict';

exports.__esModule = true;

var _manager = require('./manager');

var _manager2 = _interopRequireDefault(_manager);

var _context = require('./context');

var _context2 = _interopRequireDefault(_context);

var _scroll = require('../../utils/scroll');

var _scroll2 = _interopRequireDefault(_scroll);

var _event = require('../../utils/event');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  props: {
    // whether to show popup
    value: Boolean,
    // whether to show overlay
    overlay: Boolean,
    // overlay custom style
    overlayStyle: Object,
    // overlay custom class name
    overlayClass: String,
    // whether to close popup when click overlay
    closeOnClickOverlay: Boolean,
    // z-index
    zIndex: [String, Number],
    // prevent touchmove scroll
    preventScroll: Boolean,
    // prevent body scroll
    lockOnScroll: {
      type: Boolean,
      default: true
    }
  },

  watch: {
    value: function value(val) {
      this[val ? 'open' : 'close']();
    }
  },

  beforeMount: function beforeMount() {
    this._popupId = 'popup-' + _context2.default.plusKey('idSeed');
  },
  data: function data() {
    return {
      opened: false,
      pos: {
        x: 0,
        y: 0
      }
    };
  },


  methods: {
    recordPosition: function recordPosition(e) {
      this.pos = {
        x: e.touches[0].clientX,
        y: e.touches[0].clientY
      };
    },
    watchTouchMove: function watchTouchMove(e) {
      var pos = this.pos;

      var dx = e.touches[0].clientX - pos.x;
      var dy = e.touches[0].clientY - pos.y;
      var direction = dy > 0 ? '10' : '01';
      var el = _scroll2.default.getScrollEventTarget(e.target, this.$el);
      var scrollHeight = el.scrollHeight,
          offsetHeight = el.offsetHeight,
          scrollTop = el.scrollTop;

      var isVertical = Math.abs(dx) < Math.abs(dy);

      var status = '11';

      if (scrollTop === 0) {
        status = offsetHeight >= scrollHeight ? '00' : '01';
      } else if (scrollTop + offsetHeight >= scrollHeight) {
        status = '10';
      }

      if (status !== '11' && isVertical && !(parseInt(status, 2) & parseInt(direction, 2))) {
        e.preventDefault();
        e.stopPropagation();
      }
    },
    open: function open() {
      if (this.opened || this.$isServer) {
        return;
      }

      this.$emit('input', true);

      // 如果属性中传入了`zIndex`，则覆盖`context`中对应的`zIndex`
      if (this.zIndex !== undefined) {
        _context2.default.zIndex = this.zIndex;
      }

      if (this.overlay) {
        _manager2.default.openModal(this, {
          id: this._popupId,
          dom: this.$el,
          zIndex: _context2.default.plusKey('zIndex'),
          className: this.overlayClass,
          customStyle: this.overlayStyle
        });

        if (this.lockOnScroll) {
          document.body.classList.add('van-overflow-hidden');
        }
      }

      this.$el.style.zIndex = _context2.default.plusKey('zIndex');
      this.opened = true;

      if (this.preventScroll) {
        (0, _event.on)(document, 'touchstart', this.recordPosition);
        (0, _event.on)(document, 'touchmove', this.watchTouchMove);
      }
    },
    close: function close() {
      if (!this.opened || this.$isServer) {
        return;
      }

      this.$emit('input', false);
      this.opened = false;
      this.doAfterClose();
    },
    doAfterClose: function doAfterClose() {
      _manager2.default.closeModal(this._popupId);

      if (this.lockOnScroll) {
        document.body.classList.remove('van-overflow-hidden');
      }

      if (this.preventScroll) {
        (0, _event.off)(document, 'touchstart', this.recordPosition);
        (0, _event.off)(document, 'touchmove', this.watchTouchMove);
      }
    }
  },

  beforeDestroy: function beforeDestroy() {
    this.doAfterClose();
  }
};
