// vant/lib/tab/index.js
'use strict';

exports.__esModule = true;

var _utils = require('../utils');

var _findParent = require('../mixins/find-parent');

var _findParent2 = _interopRequireDefault(_findParent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _utils.create)({
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('div', { staticClass: "van-tab__pane", class: { 'van-tab__pane--select': _vm.index === _vm.parentGroup.curActive } }, [_vm._t("default")], 2);
  },

  name: 'van-tab',

  mixins: [_findParent2.default],

  props: {
    title: {
      type: String,
      required: true
    },
    disabled: Boolean
  },

  computed: {
    index: function index() {
      return this.parentGroup.tabs.indexOf(this);
    }
  },

  created: function created() {
    this.findParentByName('van-tabs');
    this.parentGroup.tabs.push(this);
  },
  destroyed: function destroyed() {
    this.parentGroup.tabs.splice(this.index, 1);
  }
});
