// vant/lib/goods-action-mini-btn/index.js
'use strict';

exports.__esModule = true;

var _utils = require('../utils');

var _routerLink = require('../mixins/router-link');

var _routerLink2 = _interopRequireDefault(_routerLink);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _utils.create)({
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('a', { staticClass: "van-goods-action__mini-btn van-hairline", attrs: { "href": _vm.url }, on: { "click": _vm.onClick } }, [_c('icon', { staticClass: "van-goods-action__mini-btn-icon", class: _vm.iconClass, attrs: { "info": _vm.info, "name": _vm.icon } }), _vm._t("default", [_vm._v(_vm._s(_vm.text))])], 2);
  },

  name: 'van-goods-action-mini-btn',

  mixins: [_routerLink2.default],

  props: {
    url: String,
    text: String,
    info: String,
    icon: String,
    iconClass: String
  },

  methods: {
    onClick: function onClick(event) {
      this.$emit('click', event);
      this.routerLink();
    }
  }
});
