// vant/lib/switch-cell/index.js
'use strict';

exports.__esModule = true;

var _cell = require('../cell');

var _cell2 = _interopRequireDefault(_cell);

var _switch = require('../switch');

var _switch2 = _interopRequireDefault(_switch);

var _utils = require('../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _utils.create)({
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('cell', { staticClass: "van-switch-cell", attrs: { "title": _vm.title } }, [_c('van-switch', _vm._b({ on: { "input": function input($event) {
          _vm.$emit('input', $event);
        } } }, 'van-switch', _vm.$props, false))], 1);
  },

  name: 'van-switch-cell',

  components: {
    Cell: _cell2.default,
    VanSwitch: _switch2.default
  },

  props: {
    title: String,
    value: Boolean,
    loading: Boolean,
    disabled: Boolean
  },

  watch: {
    value: function value() {
      this.$emit('change', this.value);
    }
  }
});
