// vant/lib/contact-list/index.js
'use strict';

exports.__esModule = true;

var _cell = require('../cell');

var _cell2 = _interopRequireDefault(_cell);

var _radio = require('../radio');

var _radio2 = _interopRequireDefault(_radio);

var _cellGroup = require('../cell-group');

var _cellGroup2 = _interopRequireDefault(_cellGroup);

var _radioGroup = require('../radio-group');

var _radioGroup2 = _interopRequireDefault(_radioGroup);

var _utils = require('../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _utils.create)({
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('div', { staticClass: "van-contact-list" }, [_c('radio-group', { attrs: { "value": _vm.value }, on: { "input": function input($event) {
          _vm.$emit('input', $event);
        } } }, [_c('cell-group', _vm._l(_vm.list, function (item, index) {
      return _c('cell', { key: item.id }, [_c('radio', { attrs: { "name": item.id }, on: { "click": function click($event) {
            _vm.$emit('select', item, index);
          } } }, [_c('p', { staticClass: "van-contact-list__text" }, [_vm._v(_vm._s(_vm.$t('name')) + "：" + _vm._s(item.name))]), _c('p', { staticClass: "van-contact-list__text" }, [_vm._v(_vm._s(_vm.$t('tel')) + "：" + _vm._s(item.tel))])]), _c('icon', { staticClass: "van-contact-list__edit", attrs: { "name": "edit" }, on: { "click": function click($event) {
            _vm.$emit('edit', item, index);
          } } })], 1);
    }))], 1), _c('cell', { staticClass: "van-contact-list__add van-hairline--top", attrs: { "icon": "add", "title": _vm.addText || _vm.$t('addText'), "isLink": "" }, on: { "click": function click($event) {
          _vm.$emit('add');
        } } })], 1);
  },

  name: 'van-contact-list',

  components: {
    Cell: _cell2.default,
    Radio: _radio2.default,
    CellGroup: _cellGroup2.default,
    RadioGroup: _radioGroup2.default
  },

  props: {
    value: {},
    addText: String,
    list: {
      type: Array,
      default: function _default() {
        return [];
      }
    }
  }
});
