// vant/lib/actionsheet/index.js
'use strict';

exports.__esModule = true;

var _utils = require('../utils');

var _popup = require('../mixins/popup');

var _popup2 = _interopRequireDefault(_popup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _utils.create)({
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('transition', { attrs: { "name": "van-actionsheet-float" } }, [_c('div', { directives: [{ name: "show", rawName: "v-show", value: _vm.value, expression: "value" }], staticClass: "van-actionsheet", class: { 'van-actionsheet--withtitle': _vm.title } }, [_vm.title ? _c('div', { staticClass: "van-actionsheet__header van-hairline--top-bottom" }, [_c('div', { domProps: { "textContent": _vm._s(_vm.title) } }), _c('icon', { attrs: { "name": "close" }, on: { "click": function click($event) {
          $event.stopPropagation();_vm.$emit('input', false);
        } } })], 1) : _vm._e(), !_vm.title ? _c('ul', { staticClass: "van-actionsheet__list" }, _vm._l(_vm.actions, function (item, index) {
      return _c('li', { key: index, staticClass: "van-actionsheet__item van-hairline--top", class: [item.className, { 'van-actionsheet__item--loading': item.loading }], on: { "click": function click($event) {
            $event.stopPropagation();_vm.onClickItem(item);
          } } }, [!item.loading ? [_c('span', { staticClass: "van-actionsheet__name" }, [_vm._v(_vm._s(item.name))]), item.subname ? _c('span', { staticClass: "van-actionsheet__subname" }, [_vm._v(_vm._s(item.subname))]) : _vm._e()] : _c('loading', { staticClass: "van-actionsheet__loading", attrs: { "type": "circle" } })], 2);
    })) : _vm._e(), _vm.cancelText ? _c('div', { staticClass: "van-actionsheet__item van-actionsheet__cancel van-hairline--top", domProps: { "textContent": _vm._s(_vm.cancelText) }, on: { "click": function click($event) {
          $event.stopPropagation();_vm.$emit('input', false);
        } } }) : _c('div', { staticClass: "van-actionsheet__content" }, [_vm._t("default")], 2)])]);
  },

  name: 'van-actionsheet',

  mixins: [_popup2.default],

  props: {
    value: Boolean,
    title: String,
    cancelText: String,
    actions: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    overlay: {
      type: Boolean,
      default: true
    },
    closeOnClickOverlay: {
      type: Boolean,
      default: true
    }
  },

  mounted: function mounted() {
    this.value && this.open();
  },


  methods: {
    onClickItem: function onClickItem(item) {
      if (typeof item.callback === 'function') {
        item.callback(item);
      }
    }
  }
});
